File: 0957-minimum-add-to-make-parentheses-valid/solution.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        unbalance = []
        for i in s:
            if not unbalance:
                unbalance.append(i)
            else:
                if (i == ")") and (unbalance[-1] == "("):
                    unbalance.pop()
                else:
                    unbalance.append(i)
        return len(unbalance)

File: 0957-minimum-add-to-make-parentheses-valid/solution.java
class Solution {
    public int minAddToMakeValid(String s) {
        int ret = 0;
        int balance = 0;
        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i)=='(')
                balance++;
            else
                balance--;
            if(balance==-1) {
                ret++;
                balance=0;
            }
        }
        ret += balance;
        return ret;
    }
}
